wapitiCore/attack/mod_https_redirect.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional
+from urllib.parse import urlparse, urlunparse
+import socket
+
+from httpx import RequestError
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.definitions.https_redirect import NAME, WSTG_CODE
+from wapitiCore.definitions.internal_error import WSTG_CODE as INTERNAL_ERROR_WSTG_CODE
+from wapitiCore.language.vulnerability import Messages
+from wapitiCore.net import Request
+from wapitiCore.net.response import Response
+from wapitiCore.main.log import log_red, log_orange
+
+def test_port(address: str, dest_port: int, timeout: float = None) -> bool:
```

Comment:
You should use the `@lru_cache` decorator to prevent connection tests from being repeated